### PR TITLE
Add 1.13 conformance test

### PR DESCRIPTION
**Pull request by aanm**
The 1.13 conformance test was added during the 1.13 release cycle with
a27609a3eeef6f4d77d3a32e2cf0a5682fe301f8 so we should also be testing
the v1.13 branch for regressions.
    
Fixes: fc7ae83692b1 ("Prepare for v1.14 development cycle")

### CODE REVIEW

**Review by nbusseneau — APPROVED** [review]

File: .github/workflows/conformance-datapath-v1.13.yaml [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+            encryption-node: 'true'
+            lb-mode: 'snat'
+
+          - name: '7'
+            kernel: 'bpf-next-main'
+            kube-proxy: 'none'
+            kpr: 'strict'
+            tunnel: 'disabled'
+            encryption: 'wireguard'
+            encryption-node: 'true'
```
Comment (brb) [inline .github/workflows/conformance-datapath-v1.13.yaml]: v1.13 doesn't support node encryption.

**Review by brb — CHANGES_REQUESTED** [review]
Thanks. Small comment.

File: .github/workflows/conformance-datapath-v1.13.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/conformance-datapath-v1.13.yaml]:
```
diff --git a/.github/workflows/conformance-datapath-v1.13.yaml b/.github/workflows/conformance-datapath-v1.13.yaml
index ca754262a4..4933af3a9a 100644
--- a/.github/workflows/conformance-datapath-v1.13.yaml
+++ b/.github/workflows/conformance-datapath-v1.13.yaml
@@ -165,24 +165,6 @@ jobs:
             endpoint-routes: 'true'
 
           - name: '6'
-            kernel: '6.0-main'
-            kube-proxy: 'none'
-            kpr: 'strict'
-            tunnel: 'vxlan'
-            encryption: 'wireguard'
-            encryption-node: 'true'
-            lb-mode: 'snat'
-
-          - name: '7'
-            kernel: 'bpf-next-main'
-            kube-proxy: 'none'
-            kpr: 'strict'
-            tunnel: 'disabled'
-            encryption: 'wireguard'
-            encryption-node: 'true'
-            lb-mode: 'snat'
-
-          - name: '8'
             kernel: 'bpf-next-main'
             kube-proxy: 'iptables'
             kpr: 'disabled'
```

Comment (aanm) [reply .github/workflows/conformance-datapath-v1.13.yaml]: @brb thanks, I've pushed new changes. Sounds good?

Comment (brb) [reply .github/workflows/conformance-datapath-v1.13.yaml]: Noup :-) Keep the configuration, but just delete the `encryption-node` lines.

Comment (aanm) [reply .github/workflows/conformance-datapath-v1.13.yaml]: @brb done

**Review by brb — APPROVED** [review]
Thanks. I assume you have tried running tests?

**Comment by aanm** [comment]
Tests were successful [here](https://github.com/cilium/cilium/actions/runs/4353106838/jobs/7606730105) merging

**Pull request merged**